lib/Onyx.js

```diff
@@ -1021,17 +1021,18 @@ function evictStorageAndRetry(error, onyxMethod, ...args) {
  *
  * @param {String} key
  * @param {*} value
- * @param {Boolean} hasChanged
  * @param {String} method
+ * @param {Boolean} hasChanged
+ * @param {Boolean} wasRemoved
  * @returns {Promise}
  */
-function broadcastUpdate(key, value, hasChanged, method) {
+function broadcastUpdate(key, value, method, hasChanged, wasRemoved = false) {
     // Logging properties only since values could be sensitive things we don't want to log
     Logger.logInfo(`${method}() called for key: ${key}${_.isObject(value) ? ` properties: ${_.keys(value).join(',')}` : ''}`);
 
     // Update subscribers if the cached value has changed, or when the subscriber specifically requires
     // all updates regardless of value changes (indicated by initWithStoredValues set to false).
-    if (hasChanged) {
+    if (hasChanged && !wasRemoved) {
```

Comment:
Why would we not update the cache when the key is removed?

Reply:
+1, I don't understand this

Reply:
Same as in the the other thread, the call to `remove` already drops the key from cache: https://github.com/margelo/react-native-onyx/blob/ac8faed9890bca04f889f93e0f249191ee6301b5/lib/Onyx.js#L965

That's why we have to omit updating the storage here, and instead just add the key to the list of accessed keys.

Reply:
Makes sense thanks